### ISSUE TITLE
Start/stop bots to optimise account balance use

### ISSUE THREAD

**Issue by grumpydolphin** [cyberjunky/3commas-cyber-bots#154]
It would be great for a script to monitor open trades and maximise the maximum active deals.

For example, you have 100 bots enabled and you only have enough funds to run 10.  All the bots stay active until 10 deals are opened. at this point the script disables all the remaining bots.  When deals are closed the script then enables more bots to pickup more deals.  

This sort script could allow you to maximise deals with higher funding rather that spread the funds between more bots.

**Comment by amargedon** [comment]
Hmm I understand the idea but what do you think about bots that open a deal at the same time? With 100 bots in this example, it could be two bots start a deal at the same time (or within the same interval for the script) causing actually too much funds being used (or, errors at 3C regarding insufficiënt funds). 

And bots using more funds than available should already be disabled during the processing. It could be one bot uses 20% of the funds for one deal, so if only 18% is left that bot should be disabled already. Makes the script also more complicated.

Is this request based on how you work with the bots currently? How is that going given the situations described above?

**Comment by grumpydolphin** [comment]
Yes that the problem, if deals start at the same time... or too close for the script to catch.

I used a script in the past that had a similar function.  The previous script was designed to work with futures bots.  The script monitored your margin level and stopped/started bots based upon your margin. It would have suffered from the same issue though in that multiple bots could start at the same time.
